src/huggingface_hub/hub_mixin.py

```diff
@@ -33,44 +34,45 @@ def save_pretrained(
         self,
         save_directory: str,
         config: Optional[dict] = None,
+        # NOTE: without new arguments this behavior (with push_to_hub=True), will be
+        # deprecated as well. Should we already plan to extend the "save_pretrained"
+        # function or should we let this feature die and then force users to do
+        # `model.save_pretrained(...)` and `model.push_to_hub(...)` separately ?
+        # Main argument that is missing I think is "repo_id". All other are optional but
+        # nice to have.
+        # TODO: To be discussed and to remove before merging PR
```

Comment:
Do not forget to remove.
Previously I guess the other arguments were optional because of the git-based approach.

What is the policy now ? 
Should we add all the arguments here as well (repo_id, branch, create_pr, commit_message, private...) ? Or just `repo_id` which is mandatory to push to hub ? Or just deprecate this feature ?

Reply:
The signature we have in other libraries seems to be the following (here transformers):
```
            save_directory (`str` or `os.PathLike`):
                Directory where the configuration JSON file will be saved (will be created if it does not exist).
            push_to_hub (`bool`, *optional*, defaults to `False`):
                Whether or not to push your model to the Hugging Face model hub after saving it. You can specify the
                repository you want to push to with `repo_id` (will default to the name of `save_directory` in your
                namespace).
            kwargs:
                Additional key word arguments passed along to the [`~utils.PushToHubMixin.push_to_hub`] method.
```

I think I'd aim at having a similar one here. It seems identical, with a docstring clarification of how the `repo_id` is made:
```
will default to the name of `save_directory` in your namespace
```

Reply:
Makes sense ! I'll adapt like this.

Reply:
Addressed and tested in [d039ad5](https://github.com/LysandreJik/huggingface_hub/pull/8/commits/d039ad57e14cd1b8e2a8f1e5f5aa23e213f44723).